pkg/plugins/common/kustomize/v1/scaffolds/internal/templates/config/prometheus/kustomization.go

```diff
@@ -40,6 +40,13 @@ func (f *Kustomization) SetTemplateDefaults() error {
 	return nil
 }
 
-const kustomizationTemplate = `resources:
+const kustomizationTemplate = `
+# This kustomization file config contains the required manifests to allow scrapping the metrics for Prometheus using the Prometheus operator.
+# More info: https://github.com/prometheus-operator/prometheus-operator.
+# Note that in this config, you can find the ServiceMonitor that configures the metrics endpoint
+# and the permissions (RBAC) to allow Prometheus Operator to access the ServiceMonitor.
+resources:
 - monitor.yaml
+- cluster_role.yaml
+- cluster_role_binding.yaml
```

Comment:
Add the RBAC here is broken the kustomize build: 

> running: /home/prow/go/src/sigs.k8s.io/kubebuilder/test/e2e/v3/e2e-kssw/bin/kustomize build config/default
warning: /home/prow/go/src/sigs.k8s.io/kubebuilder/test/e2e/v3/e2e-kssw/bin/kustomize build config/default failed with error: (exit status 1) 2022/07/19 15:18:30 evalsymlink failure on '/home/prow/go/src/sigs.k8s.io/kubebuilder/test/e2e/v3/e2e-kssw/config/prometheus/cluster_role.yaml' : lstat /home/prow/go/src/sigs.k8s.io/kubebuilder/test/e2e/v3/e2e-kssw/config/prometheus/cluster_role.yaml: no such file or directory
Error: accumulating resources: recursed accumulation of path '/home/prow/go/src/sigs.k8s.io/kubebuilder/test/e2e/v3/e2e-kssw/config/prometheus': accumulating resources: accumulating resources from 'cluster_role.yaml': evalsymlink failure on '/home/prow/go/src/sigs.k8s.io/kubebuilder/test/e2e/v3/e2e-kssw/config/prometheus/cluster_role.yaml' : lstat /home/prow/go/src/sigs.k8s.io/kubebuilder/test/e2e/v3/e2e-kssw/config/prometheus/cluster_role.yaml: no such file or directory


You can test it locally to see where / how we can add these RBAC.
We might need to scaffold these ones in the config/rbac instead and then with the name prometheus_clsuter_role. 

@machadovilaca

Reply:
I think I fixed the issue, but `pull-kubebuilder-e2e-k8s-1-19-16 ` failed again. I think it might have been some flakiness. wdyt? Can you rerun the test?

@camilamacedo86